src/alloc_bytes.rs

```diff
@@ -30,10 +30,13 @@ impl Clone for MiriAllocBytes {
 
 impl Drop for MiriAllocBytes {
     fn drop(&mut self) {
-        if self.layout.size() != 0 {
-            // SAFETY: Invariant, `self.ptr` points to memory allocated with `self.layout`.
-            unsafe { alloc::dealloc(self.ptr, self.layout) }
-        }
+        let alloc_layout = if self.layout.size() == 0 {
```

Comment:
Can't you store the changed layout with size 1 in the `layout`? Then you don't need to duplicate the logic here.

Reply:
I initially did it where if `size == 0` was requested I simply changed `layout.size() == 1`, but this surprisingly failed all tests:
Here I thought it would be better to ensure *allocating* (and deallocating) at least one byte, but leave the remaining part of a `MiriAllocBytes` object with `size == 0` intact.